pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -647,6 +720,24 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) e
 
 		log.Infof("Handle service %s in namespace %s", svc.Name, svc.Namespace)
 
+		if directEDS {
```

Comment:
???
This is why I don’t so many environment variables. Something goes wrong and we suggest setting this obscure environment variable for direct eds. And then normal routing/services stop working completely (ie the logic in this if block, that ignores the service updates)

Reply:
I'll add more comments - I thought it is clear. It doesn't ignore the service updates - ConfigUpdate(true) is called at the end. What it does is avoids the useless 'convertService' - the 'ClearCache' handler is ignoring the parameters, so we waste CPU/memory.